<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1042</width>
    <height>558</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="styleSheet">
   <string notr="true"/>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QCustomPlot" name="customPlot" native="true">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>500</width>
      <height>500</height>
     </rect>
    </property>
   </widget>
   <widget class="QGroupBox" name="data_groupBox">
    <property name="geometry">
     <rect>
      <x>520</x>
      <y>10</y>
      <width>250</width>
      <height>161</height>
     </rect>
    </property>
    <property name="title">
     <string>Data</string>
    </property>
    <layout class="QGridLayout" name="gridLayout">
     <item row="4" column="0">
      <widget class="QPushButton" name="add_samples">
       <property name="font">
        <font>
         <pointsize>10</pointsize>
         <italic>false</italic>
         <bold>true</bold>
        </font>
       </property>
       <property name="text">
        <string>Add Samples</string>
       </property>
      </widget>
     </item>
     <item row="4" column="2">
      <widget class="QLabel" name="pos_label">
       <property name="font">
        <font>
         <pointsize>8</pointsize>
         <bold>false</bold>
        </font>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
     <item row="0" column="0" colspan="2">
      <widget class="QLabel" name="num_class_label">
       <property name="font">
        <font>
         <pointsize>10</pointsize>
         <bold>true</bold>
        </font>
       </property>
       <property name="text">
        <string>Number of Classes:</string>
       </property>
      </widget>
     </item>
     <item row="0" column="2">
      <widget class="QLineEdit" name="class_lineEdit"/>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="labels">
       <property name="font">
        <font>
         <pointsize>10</pointsize>
         <bold>true</bold>
        </font>
       </property>
       <property name="text">
        <string>Labels:</string>
       </property>
      </widget>
     </item>
     <item row="1" column="2">
      <widget class="QComboBox" name="label_comboBox">
       <item>
        <property name="text">
         <string>---------------</string>
        </property>
       </item>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QGroupBox" name="results_groupBox">
    <property name="geometry">
     <rect>
      <x>520</x>
      <y>170</y>
      <width>250</width>
      <height>81</height>
     </rect>
    </property>
    <property name="title">
     <string>Training Results</string>
    </property>
    <layout class="QGridLayout" name="gridLayout_2">
     <item row="0" column="0">
      <widget class="QLabel" name="cycles_label">
       <property name="font">
        <font>
         <pointsize>10</pointsize>
         <bold>true</bold>
        </font>
       </property>
       <property name="text">
        <string>Cycles (Epoch):</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QLabel" name="cycles_value">
       <property name="font">
        <font>
         <pointsize>10</pointsize>
        </font>
       </property>
       <property name="text">
        <string>0</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QLabel" name="error_value">
       <property name="font">
        <font>
         <pointsize>10</pointsize>
        </font>
       </property>
       <property name="text">
        <string>0</string>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="error_label">
       <property name="font">
        <font>
         <pointsize>10</pointsize>
         <bold>true</bold>
        </font>
       </property>
       <property name="text">
        <string>Error:</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QGroupBox" name="settings_groupBox">
    <property name="geometry">
     <rect>
      <x>790</x>
      <y>10</y>
      <width>244</width>
      <height>161</height>
     </rect>
    </property>
    <property name="title">
     <string>Network Settings</string>
    </property>
    <layout class="QGridLayout" name="gridLayout_3">
     <item row="2" column="0">
      <widget class="QLabel" name="max_error_label">
       <property name="font">
        <font>
         <pointsize>10</pointsize>
         <bold>true</bold>
        </font>
       </property>
       <property name="text">
        <string>Maximum Error:</string>
       </property>
      </widget>
     </item>
     <item row="4" column="0">
      <widget class="QLabel" name="output_label">
       <property name="font">
        <font>
         <pointsize>10</pointsize>
         <bold>true</bold>
        </font>
       </property>
       <property name="text">
        <string>Output Layer:</string>
       </property>
      </widget>
     </item>
     <item row="4" column="1">
      <widget class="QLineEdit" name="output_value"/>
     </item>
     <item row="2" column="1">
      <widget class="QLineEdit" name="max_error_value"/>
     </item>
     <item row="1" column="1">
      <widget class="QLineEdit" name="learning_value"/>
     </item>
     <item row="3" column="0">
      <widget class="QLabel" name="hidden_label">
       <property name="font">
        <font>
         <pointsize>10</pointsize>
         <bold>true</bold>
        </font>
       </property>
       <property name="text">
        <string>Hidden Layer:</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QLineEdit" name="epoch_value"/>
     </item>
     <item row="3" column="1">
      <widget class="QLineEdit" name="hidden_value"/>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="learning_label">
       <property name="font">
        <font>
         <pointsize>10</pointsize>
         <bold>true</bold>
        </font>
       </property>
       <property name="text">
        <string>Learning Rate:</string>
       </property>
      </widget>
     </item>
     <item row="0" column="0">
      <widget class="QLabel" name="epoch_label">
       <property name="font">
        <font>
         <pointsize>10</pointsize>
         <bold>true</bold>
        </font>
       </property>
       <property name="text">
        <string>Maximum Epoch:</string>
       </property>
      </widget>
     </item>
     <item row="5" column="0">
      <widget class="QLabel" name="normalize_label">
       <property name="font">
        <font>
         <pointsize>10</pointsize>
         <bold>true</bold>
        </font>
       </property>
       <property name="text">
        <string>Normalize:</string>
       </property>
      </widget>
     </item>
     <item row="5" column="1">
      <widget class="QCheckBox" name="normalize_check_box">
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>790</x>
      <y>170</y>
      <width>241</width>
      <height>81</height>
     </rect>
    </property>
    <property name="title">
     <string>Weights</string>
    </property>
    <widget class="QLabel" name="w1_label">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>31</width>
       <height>16</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>10</pointsize>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string>w1:</string>
     </property>
    </widget>
    <widget class="QLabel" name="w2_label">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>40</y>
       <width>31</width>
       <height>16</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>10</pointsize>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string>w2:</string>
     </property>
    </widget>
    <widget class="QLabel" name="w3_label">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>60</y>
       <width>31</width>
       <height>16</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>10</pointsize>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string>w3:</string>
     </property>
    </widget>
    <widget class="QLabel" name="w1_value">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>20</y>
       <width>181</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>0</string>
     </property>
    </widget>
    <widget class="QLabel" name="w2_value">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>40</y>
       <width>171</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>0</string>
     </property>
    </widget>
    <widget class="QLabel" name="w3_value">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>60</y>
       <width>171</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>0</string>
     </property>
    </widget>
   </widget>
   <widget class="QCustomPlot" name="error_plot" native="true">
    <property name="geometry">
     <rect>
      <x>520</x>
      <y>270</y>
      <width>511</width>
      <height>241</height>
     </rect>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1042</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuProcess">
    <property name="title">
     <string>Process</string>
    </property>
    <widget class="QMenu" name="menuSingle_Layer">
     <property name="title">
      <string>Single Layer</string>
     </property>
     <addaction name="actionRandom"/>
     <addaction name="actionTrain_Delta"/>
     <addaction name="actionTest"/>
    </widget>
    <widget class="QMenu" name="menuMulti_Layer">
     <property name="title">
      <string>Multi Layer</string>
     </property>
     <addaction name="actionRandom_2"/>
     <addaction name="actionTrain_Delta_2"/>
     <addaction name="actionTrain_Delta_with_Moment"/>
     <addaction name="actionTest_2"/>
    </widget>
    <addaction name="menuSingle_Layer"/>
    <addaction name="menuMulti_Layer"/>
    <addaction name="actionRestart"/>
   </widget>
   <addaction name="menuProcess"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionRandom">
   <property name="text">
    <string>Random</string>
   </property>
  </action>
  <action name="actionTrain_Delta">
   <property name="text">
    <string>Train (Delta)</string>
   </property>
  </action>
  <action name="actionRandom_2">
   <property name="text">
    <string>Random</string>
   </property>
  </action>
  <action name="actionTrain_Delta_2">
   <property name="text">
    <string>Train (Delta)</string>
   </property>
  </action>
  <action name="actionTrain_Delta_with_Moment">
   <property name="text">
    <string>Train (Delta with Moment)</string>
   </property>
  </action>
  <action name="actionTest">
   <property name="text">
    <string>Test</string>
   </property>
  </action>
  <action name="actionTest_2">
   <property name="text">
    <string>Test</string>
   </property>
  </action>
  <action name="actionRestart">
   <property name="text">
    <string>Restart</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QCustomPlot</class>
   <extends>QWidget</extends>
   <header>qcustomplot.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
